jina/executors/evaluators/decorators.py

```diff
@@ -0,0 +1,19 @@
+"""Decorators and wrappers designed for wrapping :class:`BaseExecutor` functions. """
+
+from functools import wraps
+from typing import Callable
+
+
+def as_aggregator(func: Callable) -> Callable:
+    """Mark a function so that it keeps track of the number of documents evaluated and a running sum
+    to have always access to average value
+    """
+
+    @wraps(func)
+    def arg_wrapper(self, *args, **kwargs):
+        f = func(self, *args, **kwargs)
+        self.num_documents += 1
+        self.sum += f
+        return f
+
+    return arg_wrapper
```

Comment:
I don't see how this decorator can be used in general. maybe it's better to just write them inline?

Reply:
I would remove this part until we have clear vision on how we want the user to interact with `averages` and `running metrics`.  How would a user `reset` the count to 0? I would personally leave the metrics part on the client side

Reply:
>how we want the user to interact with averages and running metrics

kill the flow/pod and restart

Reply:
Won't it affect the user experience as seen as a user may want to keep the Flow running and evaluate just few requests and then keep searching and then evaluating and so....?

Would you agree to remove this part and handle it in a different PR?